frontend/src/app/password-reset/password-reset-confirm.component.html

```diff
@@ -0,0 +1,51 @@
+<div class="container">
+  <h2>Potwierdź zmianę hasła</h2>
+  <form (ngSubmit)="onSubmit()" [formGroup]="resetForm">
+    <div  formGroupName="passwords">
+      <div class="form-group">
+        <label for="password">Podaj nowe hasło</label>
+        <input
+          type="password"
+          id="password"
+          formControlName="password"
+          class="form-control form-control-lg"
+          [ngClass]="{'is-invalid': resetForm.get('passwords.password').invalid && resetForm.get('passwords.password').touched}"
```

Comment:
I have two suggestions here. Please also bear in mind my comment about MVC pattern, which applies somehow here as well.

1. there is a repeatable logic throughout the template that you use for checking if input is invalid && touched. Move it to component method, e.g:
```
isFormInputInvalid(inputStringId: string) : boolean {
  return resetForm.get(inputStringId).invalid && resetForm.get(inputStringId).touched;
}
```
2. There is also a repeatable logic for your ngClass binding. I would use a method defined on component as well, e.g. 
```
getClassForFormInput(inputStringId: string) : any {
  return {
    'is-invalid': isFormInputInvalid(inputStringId)
  }
}
```
Please note that this solution is easily extendable and we can quickly define a logic for adding another class to the input here. It also uses a solution from point 1.

Also, bear in mind that programmers are lazy creatures by nature and don't like to repeat themselves, so that's why we keep it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) ;)

Reply:
That's valid comment, if it will work I'm :+1:

Reply:
I've added `isFormInputInvalid` method in component and I use it in the template in both *ngIf and ngClass directives. I think we don't need the second method for adding classes - we probably won't add another class to inputs in this form.